engine/itest/src/test/java/org/opennms/oce/engine/itest/ResumeServiceTest.java

```diff
@@ -0,0 +1,158 @@
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.oce.engine.itest;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.opennms.oce.datasource.api.Alarm;
+import org.opennms.oce.datasource.api.Incident;
+import org.opennms.oce.datasource.api.InventoryObject;
+import org.opennms.oce.datasource.api.ResourceKey;
+import org.opennms.oce.datasource.api.Severity;
+import org.opennms.oce.driver.test.MockAlarmBuilder;
+import org.opennms.oce.driver.test.MockInventory;
+import org.opennms.oce.driver.test.TestDriver;
+import org.opennms.oce.engine.api.EngineFactory;
+import org.opennms.oce.engine.cluster.ClusterEngineFactory;
+import org.opennms.oce.engine.temporal.TimeSliceEngineFactory;
+import org.opennms.oce.engine.topology.TopologyEngineFactory;
+
+@RunWith(Parameterized.class)
+public class ResumeServiceTest {
+
+    @Parameterized.Parameters(name = "{index}: engine({0})")
+    public static Iterable<Object[]> data() {
+        return Arrays.asList(new Object[][] { { new TimeSliceEngineFactory() },
+                { new ClusterEngineFactory() },
+                { new TopologyEngineFactory() }
+        });
+    }
+
+    private List<InventoryObject> inventory = MockInventory.SAMPLE_NETWORK;
+
+    private final EngineFactory factory;
+    private TestDriver driver;
+
+    public ResumeServiceTest(EngineFactory factory) {
+        this.factory = Objects.requireNonNull(factory);
+    }
+
+    @Before
+    public void setUp() {
+        driver = TestDriver
+                .builder()
+                .withEngineFactory(factory)
+                .build();
+    }
+
+    @Test
+    public void resumeServiceTest() {
+        final List<Alarm> alarms = new ArrayList<>();
+        // Fail all ports on all cards of node: n1
+        alarms.addAll(new MockAlarmBuilder()
+                .withId("a1")
+                .withResourceKey(new ResourceKey("Port,n1-c1-p1"))
+                .withEvent(SECONDS.toMillis(1), Severity.MAJOR)
+                .build());
+        alarms.addAll(new MockAlarmBuilder()
+                .withId("a2")
+                .withResourceKey(new ResourceKey("Port,n1-c1-p2"))
+                .withEvent(SECONDS.toMillis(31), Severity.MAJOR)
+                .build());
+        
+        final List<Incident> baseIncidents = driver.run(alarms, inventory);
```

Comment:
I'd add an assertion that the base run should generate at least 1 incident, otherwise the test may not be verifying the behavior as intended.